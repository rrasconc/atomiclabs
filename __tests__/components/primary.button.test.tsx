import { render, screen } from '@testing-library/react-native';

import { PrimaryButton } from '../../components/Primary.Button';

describe('<PrimaryButton loading />', () => {
  let component: any = null;
  beforeEach(() => {
    component = render(<PrimaryButton loading text="Enviar" />);
  });

  it('renders correctly', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('shows activity indicator', () => {
    expect(screen.queryByText(/enviar/i)).toBeNull();
  });
});

describe('<PrimaryButton />', () => {
  let component: any = null;
  beforeEach(() => {
    component = render(<PrimaryButton text="Enviar" />);
  });

  it('renders correctly', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('shows text', () => {
    expect(screen.getByText(/enviar/i)).toBeDefined();
  });
});
