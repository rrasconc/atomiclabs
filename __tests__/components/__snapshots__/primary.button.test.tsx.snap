// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrimaryButton /> renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FA4D09",
        "borderRadius": 100,
        "justifyContent": "center",
        "opacity": 1,
        "padding": 32.14285714285714,
        "paddingHorizontal": 171.42857142857142,
        "width": "auto",
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
            "fontSize": 38.57142857142857,
          },
          {
            "fontWeight": "500",
          },
        ]
      }
    >
      Enviar
    </Text>
  </View>
</View>
`;

exports[`<PrimaryButton loading /> renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FA4D09",
        "borderRadius": 100,
        "justifyContent": "center",
        "opacity": 1,
        "padding": 32.14285714285714,
        "paddingHorizontal": 171.42857142857142,
        "width": "auto",
      }
    }
  >
    <ActivityIndicator
      color="#FFFFFF"
    />
  </View>
</View>
`;
